### PR TITLE
make: Fix Makefile to conform to the current convention

### PR DESCRIPTION
**Pull request by MaciejPurski**
JIRA: DTR-56

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,82 +1,32 @@
...
-	
 .PHONY: clean
 clean:
 	@echo "rm -rf $(BUILD_DIR)"
 
 ifneq ($(filter clean,$(MAKECMDGOALS)),)
 	$(shell rm -rf $(BUILD_DIR))
 endif
+
+T1 := $(filter-out clean all,$(MAKECMDGOALS))
```
Comment (nalajcie) [inline Makefile:27]: `T1` not used, safe to delete

File: Makefile [hunk 2]
```diff
@@ -1,82 +1,32 @@
...
-	@mkdir -p $(@D)
-	@(printf "STR %-24s\n" "$(@F)")
-	$(SIL)$(STRIP) -o $@ $<
-	
-all: $(PREFIX_PROG_STRIPPED)posixsrv
-
-$(PREFIX_PROG)posixsrv: $(addprefix $(PREFIX_O), event.o pipe.o pty.o special.o tmpfile.o posixsrv.o)
-	$(LINK)
-	
 .PHONY: clean
```
Comment (nalajcie) [inline Makefile:19]: I think we can delete `clean` due to https://github.com/phoenix-rtos/phoenix-rtos-build/blob/4c9653393f8d2974e2c084275dabd175243cfc94/build.sh#L91

Comment (MaciejPurski) [reply Makefile]: Done.

File: Makefile (hunk 1 above)
Comment (MaciejPurski) [reply Makefile]: Done.

**Review by nalajcie — APPROVED** [review]

**Pull request merged**